components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/IdPInitLogoutRequestProcessor.java

```diff
@@ -70,6 +74,26 @@ public SAMLSSOReqValidationResponseDTO process(String sessionId, QueryParamDTO[]
 
         init(queryParamDTOs);
 
+        DiagnosticLog.DiagnosticLogBuilder finalizeDiagLogBuilder = null;
+        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+            // Initialize finalizeDiagLogBuilder here to avoid initializing it in every if condition.
+            finalizeDiagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                    SAML_INBOUND_SERVICE, SAMLSSOConstants.LogConstants.ActionIDs.PROCESS_SAML_LOGOUT);
+            DiagnosticLog.DiagnosticLogBuilder initializeDiagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                    SAML_INBOUND_SERVICE, SAMLSSOConstants.LogConstants.ActionIDs.PROCESS_SAML_LOGOUT);
+            initializeDiagLogBuilder.resultMessage("Processing IDP initiated logout request.")
+                    .logDetailLevel(DiagnosticLog.LogDetailLevel.APPLICATION)
+                    .resultStatus(DiagnosticLog.ResultStatus.SUCCESS)
+                    .inputParam("server url", serverURL);
```

Comment:
shouldn't we add the sp details as input params? wdyt?

Reply:
The SP details are not available here. How ever we can get those details by fetching them from the DB. Since it's a costly operation and also we are showing the issuer value of the SAML application, we may not need to add SP details here